utils/src/main/java/io/atomix/utils/serializer/serializers/ByteBufferSerializer.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Copyright 2016-present Open Networking Foundation
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.atomix.utils.serializer.serializers;
+
+import com.esotericsoftware.kryo.Kryo;
+import com.esotericsoftware.kryo.Serializer;
+import com.esotericsoftware.kryo.io.Input;
+import com.esotericsoftware.kryo.io.Output;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+
+public class ByteBufferSerializer extends Serializer<ByteBuffer> {
+
+  @Override
+  public void write(final Kryo kryo, final Output output, final ByteBuffer object) {
+    output.writeBoolean(object.isDirect());
+    output.writeBoolean(ByteOrder.LITTLE_ENDIAN.equals(object.order()));
+    output.writeInt(object.remaining());
+    for (int i = object.position(); i < object.capacity(); i++) {
+      output.writeByte(object.get(i));
+    }
+  }
+
+  @Override
+  public ByteBuffer read(final Kryo kryo, final Input input, final Class<ByteBuffer> type) {
+    final boolean isDirect = input.readBoolean();
+    final boolean isLittleEndian = input.readBoolean();
+    final int capacity = input.readInt();
+    final ByteBuffer buffer;
+
+    if (isDirect) {
+      buffer = ByteBuffer.allocateDirect(capacity);
```

Comment:
Not sure if this makes really sense. Since we use these kyro serializer/deserializer it is already copied anyway in the java heap right?

Reply:
I mean direct buffer is only useful if I for example receive the bytes from a channel and directly write it to file or read it from file and directly send over the network, without copying the data into the java heap.

Reply:
I figured I would add it since it's minimal effort and we can decide what to do in the future. 

As a side note, Kryo also supports direct buffers, we just haven't configured it for it yet. See [kryo#bytebuffers](https://github.com/EsotericSoftware/kryo#bytebuffers). I'm not sure what the optimal usage would be here.